src/mlpack/core/data/load_arff_impl.hpp

```diff
@@ -103,7 +105,7 @@ void LoadARFF(const std::string& filename,
           // `origDimType` string here instead (which has not had ::tolower used
           // on it).
           types.push_back(true);
-          boost::trim_if(origDimType,
+	  trim_if(origDimType,
```

Comment:
I am not sure of removing boost like this. You need to write a function that trim according to these characters.
It should not be hard, let me know if you need any help, basically you need to create another trim function `trim_if` that takes a lambda function as a parameter. Then you call back the lambda parameter inside the `trim_if` and recover the char that need to be used in the trimming process